### ISSUE TITLE
Editor crash on Create New Resource

### ISSUE THREAD

**Issue by Tony-Klink** [BobbyAnguelov/Esoterica#8]
This happens for:
 - Animation Bone Mask
 - Animation Clip
 - Animation Skeleton
 - Physics Ragdoll
```
TVector<ResourceDatabase::ResourceEntry*> const& ResourceDatabase::GetAllResourcesOfType( ResourceTypeID typeID ) const
    {
        EE_ASSERT( m_pTypeRegistry->IsRegisteredResourceType( typeID ) );
        return m_resourcesPerType.at( typeID ); // <- Fails with "invalid hash_map<K, T> key" error
    }
```
[2022-08-31 13-07-02.webm](https://user-images.githubusercontent.com/1422270/187655003-04153924-ecae-4785-8457-471f93a6d382.webm)

**Comment by BobbyAnguelov** [comment]
Thanks for the report. That's super strange though, and I cant repro it. There is the potential for that widget to crash but it's super unlikely. 

I added an early out for that code in the only "valid" case for this to crash. Can you try it and see if its still crashing for you?

**Comment by Tony-Klink** [comment]
Thank you for the response.
Just checked your commit. In `RefreshResourceList` assertions are passed without errors for resourceTypeID 'skel', then GetAllResourcesOfType fails with same reason.

**Comment by BobbyAnguelov** [comment]
Hm... this is super strange, especially since I cant repro it locally. I'll keep digging around and see if I can figure it out or worst case, I'll add more assert to try to figure out what's going on.

**Comment by BobbyAnguelov** [comment]
Okay, there's something really wrong with creating a new ragdoll and another issue I completely missed when refactoring the workspace.

**Comment by BobbyAnguelov** [comment]
Okay, I fixed the crash. I'm an idiot and never tested running the editor with an empty data folder. I fixed a whole bunch of other issues too and made it a lot harder to fuck something up. That said, I had to completely disable the entity/map editor temporarily as I was mid refactor when I jumped into fixing this.

I also heavily reworked the ragdoll editor so it should work quite well (also a bunch of dumb issues there)
